llvm/lib/Target/NVPTX/NVPTXISelLowering.cpp

```diff
@@ -2619,7 +2595,7 @@ NVPTXTargetLowering::LowerOperation(SDValue Op, SelectionDAG &DAG) const {
   case ISD::BUILD_VECTOR:
     return LowerBUILD_VECTOR(Op, DAG);
   case ISD::BITCAST:
-    return LowerBITCAST(Op, DAG);
+    return SDValue();
```

Comment:
Can you remove this case entirely? You might need to remove any custom setOperationAction entries for BITCAST?

Reply:
I removed the case and the `setOperationAction` but when I did that it started failing the second test case and applying the stack-based bitcast. I am unsure why it did that so I will look into it further. Unless you have any ideas?

Reply:
DAGTypeLegalizer::ExpandRes_BITCAST already does what you're after (expand bitcast of vector to scalar integer) in a more general form - maybe start there?

Reply:
Hi, I had a look at `DAGTypeLegalizer::ExpandRes_BITCAST` it is close to what I want but it emits the store/load.
I managed to get the code working for the NVPTX test case by removing the `setOperationAction`. Please will look at the newer code and see if the idea is on the right track?